assets/hdfs_append/index.js

```diff
@@ -21,6 +21,68 @@ function getClient(context, config, type) {
     return context.foundation.getConnection(clientConfig);
 }
 
+// Records the name of the offending file with a detected corrupt block in `appendErrors`
+function _recordFileError(name, errorLog) {
+    let newFilename = '';
+    // Get the original file name from the error message. If there have been no previous errors
+    // for the given file, this will result in a bogus filename that will trigger the second
+    // block of the following checks. Otherwise, the third block will be used to get a new name
+    const fileBase = name
+        .split('.')
+        .reverse()
+        .splice(1)
+        .reverse()
+        .join('.');
```

Comment:
This is actually doing something a bit tricky here for the check it precedes and is used specifically to find any subsequent append errors a worker encounters. In the event of a second (or any subsequent) append error a worker encounters, if the worker is writing to a file other than the one that previously had an append error, this will strip off the end of the filename (which I am assuming will always be the worker number that is appended to the TS host name or IP) and trigger the second condition [here](https://github.com/terascope/hdfs-assets/pull/11/files/b1ae23226f69b86007d220fa5c793bc5cc80fa83#diff-4d3bfc5cbbc070ec9b480b4e1cfdc8feR43). If this is just a subsequent error on the same file, this sequence will strip off the number this processor appended to the filename originally to keep track of the number of append errors and trigger the [third conditional](https://github.com/terascope/hdfs-assets/pull/11/files/b1ae23226f69b86007d220fa5c793bc5cc80fa83#diff-4d3bfc5cbbc070ec9b480b4e1cfdc8feR56). 

Also, this will work both with file names that include periods as well file names that don't include  periods.

Reply:
I see, I thought it was just getting the base path. You should be able to do this with simpler logic without dealing with arrays and reverses as all you're trying to do is remove the last element after the final `.`. Look at String.lastIndexOf and then simply chop that off the string.

Reply:
Cleaned this up with `String.lastIndexOf`